proposals/extensions.md

```diff
@@ -50,6 +50,48 @@ receiver_parameter // add
 Extension declarations shall only be declared in non-generic, non-nested static classes.  
 It is an error for a type to be named `extension`.  
 
+### Declaration spaces
+
+Section [7.3 Declarations](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/basic-concepts.md#73-declarations) in the C# Standard is updated as follows (additions in **bold non-italic**):
+
+> ...
+> 
+> There are several different types of declaration spaces, as described in the following.
+> 
+> ...
+> 
+> - Each method declaration, property declaration, property accessor declaration, indexer declaration, indexer accessor declaration, operator declaration, instance constructor declaration, anonymous function, and local function creates a new declaration space called a ***local variable declaration space***. Names are introduced into this declaration space through formal parameters (*fixed_parameter*s and *parameter_array*s) and *type_parameter*s. The set accessor for a property or an indexer introduces the name `value` as a formal parameter. 
+**An *extension_declaration* introduces to each member declaration directly contained within it its *type_parameter_list* as type parameters and the *identifier*, if any, of its *receiver_parameter* as a formal parameter.**
```

Comment:
Do we expect the following error and, if so, what rules requires it?
```
    extension<T>(long[] p)
    {
        void T(){} // error CS0102: The type 'Extensions.extension<T>(long[])' already contains a definition for 'T'
    }
```

Reply:
Not an error. From a declaration space point of view it is equivalent to

``` c#
public void T<T>() { }
```

because, per the new rule, the type parameter `T` is added to the local variable declaration space of the member `T`.

Reply:
> Not an error. From a declaration space point of view it is equivalent to ...

Consider adding an example to make this behavior explicit.